src/ui_interface.rs

```diff
@@ -243,7 +243,12 @@ pub fn set_peer_option(id: String, name: String, value: String) {
 
 #[inline]
 pub fn using_public_server() -> bool {
-    crate::get_custom_rendezvous_server(get_option_("custom-rendezvous-server")).is_empty()
+    let key_check: Option<&'static str> = option_env!("RS_PUB_KEY_VAL");
```

Comment:
Honestly, `crate::get_custom_rendezvous_server(get_option_("custom-rendezvous-server")).is_empty()` has bug, you last change fixed the bug, but we need to get rid of duplication.

Reply:
I added a default key option and used that, should resolve the issue